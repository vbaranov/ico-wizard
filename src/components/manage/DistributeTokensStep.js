import React from 'react'
import { Link } from 'react-router-dom'
import classNames from 'classnames'

export const DistributeTokensStep = ({ disabled, handleClick }) => (
  <div className="steps-content container">
    <div className="about-step">
      <div className="swal2-icon swal2-info warning-logo">!</div>
      <p className="title">Distribute reserved tokens</p>
      <p className="description">
        Reserved tokens distribution is the last step of the crowdsale before finalization.
        You can make it after the end of the last tier or if hard cap is reached. If you reserved more then 100
        addresses for your crowdsale, the distribution will be executed in batches with 100 reserved addresses per
        batch. Amount of batches is equal to amount of transactions
      </p>
      <Link to='#' onClick={() => !disabled ? handleClick(100) : undefined}>
        <span className={classNames(
          'button',
          {
            'button_disabled': disabled,
            'button_fill': !disabled
          }
        )}>
          Distribute tokens
        </span>
      </Link>
    </div>
  </div>
)
