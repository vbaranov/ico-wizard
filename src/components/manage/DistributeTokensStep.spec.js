import React from 'react'
import { StaticRouter } from 'react-router'
import { DistributeTokensStep } from './DistributeTokensStep'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'

configure({ adapter: new Adapter() })

describe('DistributeTokensStep', () => {
  it('should render the component with active button', () => {
    const distributeTokensStateParams = {
      disabled: false,
      handleClick: jest.fn()
    }

    expect(renderer.create(
      <StaticRouter location="testLocation" context={{}}>
        <DistributeTokensStep {...distributeTokensStateParams} />
      </StaticRouter>
    ).toJSON()).toMatchSnapshot()
  })

  it('should render the component with disabled button', () => {
    const distributeTokensStateParams = {
      disabled: true,
      handleClick: jest.fn()
    }

    expect(renderer.create(
      <StaticRouter location="testLocation" context={{}}>
        <DistributeTokensStep {...distributeTokensStateParams} />
      </StaticRouter>
    ).toJSON()).toMatchSnapshot()
  })

  it('should call handleClick', () => {
    const distributeTokensStateParams = {
      disabled: false,
      handleClick: jest.fn()
    }

    const wrapper = mount(
      <StaticRouter location="testLocation" context={{}}>
        <DistributeTokensStep {...distributeTokensStateParams} />
      </StaticRouter>
    )

    const button = wrapper.find('Link').at(0)

    button.simulate('click')

    expect(distributeTokensStateParams.handleClick).toHaveBeenCalled()
  })

  it('should not call handleClick', () => {
    const distributeTokensStateParams = {
      disabled: true,
      handleClick: jest.fn()
    }

    const wrapper = mount(
      <StaticRouter location="testLocation" context={{}}>
        <DistributeTokensStep {...distributeTokensStateParams} />
      </StaticRouter>
    )

    const button = wrapper.find('Link').at(0)

    button.simulate('click')

    expect(distributeTokensStateParams.handleClick).toHaveBeenCalledTimes(0)
  })
})
