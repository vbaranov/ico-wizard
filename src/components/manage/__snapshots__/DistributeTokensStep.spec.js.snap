// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistributeTokensStep should render the component with active button 1`] = `
<div
  className="steps-content container"
>
  <div
    className="about-step"
  >
    <div
      className="swal2-icon swal2-info warning-logo"
    >
      !
    </div>
    <p
      className="title"
    >
      Distribute reserved tokens
    </p>
    <p
      className="description"
    >
      Reserved tokens distribution is the last step of the crowdsale before finalization. You can make it after the end of the last tier or if hard cap is reached. If you reserved more then 100 addresses for your crowdsale, the distribution will be executed in batches with 100 reserved addresses per batch. Amount of batches is equal to amount of transactions
    </p>
    <a
      href="/#"
      onClick={[Function]}
    >
      <span
        className="button button_fill"
      >
        Distribute tokens
      </span>
    </a>
  </div>
</div>
`;

exports[`DistributeTokensStep should render the component with disabled button 1`] = `
<div
  className="steps-content container"
>
  <div
    className="about-step"
  >
    <div
      className="swal2-icon swal2-info warning-logo"
    >
      !
    </div>
    <p
      className="title"
    >
      Distribute reserved tokens
    </p>
    <p
      className="description"
    >
      Reserved tokens distribution is the last step of the crowdsale before finalization. You can make it after the end of the last tier or if hard cap is reached. If you reserved more then 100 addresses for your crowdsale, the distribution will be executed in batches with 100 reserved addresses per batch. Amount of batches is equal to amount of transactions
    </p>
    <a
      href="/#"
      onClick={[Function]}
    >
      <span
        className="button button_disabled"
      >
        Distribute tokens
      </span>
    </a>
  </div>
</div>
`;
